src/Profiles.ts

```diff
@@ -104,15 +105,40 @@ export class Profiles { // Processing stops if there are no profiles detected
         return this.allProfiles;
     }
 
-    public validateUrl = (newUrl: string) => {
+    public validateAndParseUrl = (newUrl: string): IUrlValidator => {
+
+        let url: URL;
+        const validProtocols: string[] = ["https"];
+        const DEFAULT_HTTPS_PORT: number = 443;
+
+        const validationResult: IUrlValidator = {
+            valid: false,
+            host: null,
+            port: null
+        };
+
         try {
             url = new URL(newUrl);
         } catch (error) {
-            return false;
+            return validationResult;
+        }
+
+        // overkill with only one valid protocol, but we may expand profile types and protocols in the future?
+        if (!validProtocols.some((validProtocol: string) => url.protocol.includes(validProtocol))) {
+            return validationResult;
         }
-        validURL = url.hostname;
-        validPort = Number(url.port);
-        return url.port ? true : false;
+
+        // if port is empty, set https defaults
+        if (!url.port.trim()) {
+            validationResult.port = DEFAULT_HTTPS_PORT;
+        }
+        else {
+            validationResult.port = Number(url.port);
```

Comment:
Will this be a valid url `https://10.142.0.23:65536`?

Reply:
> Will this be a valid url `https://10.142.0.23:65536`?

I tested this one and it is not considered a valid URL